### ISSUE TITLE
Docker Inside Docker

### ISSUE THREAD

**Issue by isreehari** [remoteinterview/compilebox#60]
We have app and which will spin the short term (short span) docker containers. Right now, it runs in Ubunut16.04 server (VM) and we installed docker, and nodejs in same server. We have nodejs app which runs in same server so whenever a request comes in, then the nodejs app will spin up a docker container and execute a user input inside the docker container. Once after the docker finish its job or if it runs out of admin defined resources then the docker container will be forcefully killed (docker kill) and removed (docker rm).

Now my question is, is it best practices to run the Ubunte16.04 docker container and run nodjes app and the short term docker containers inside the Ubunuter16.04 docker container.

In short run a docker inside other docker container.

![](https://i.stack.imgur.com/qCxL5.jpg)
